lib/iris/cube.py

```diff
@@ -1700,16 +1701,11 @@ def data(self):
         if self._numpy_array is None:
             try:
                 data = self._dask_array.compute()
-                mask = np.isnan(data)
-                if data.dtype != self.dtype:
-                    data = data.astype(self.dtype)
-                    self.dtype = None
-                if np.all(~mask):
-                    self._numpy_array = data
-                else:
-                    fv = self.fill_value
-                    self._numpy_array = ma.masked_array(data, mask=mask,
-                                                        fill_value=fv)
+                self._numpy_array = array_nans_to_masked(data,
+                                                         self.fill_value,
+                                                         self.dtype)
+                self.dtype = None
```

Comment:
@pp-mo I'm expecting you to have stomped over this with your PR, but hopefully they align in logic ...